local_costmap:
  global_frame: odom
  robot_base_frame: base_link
  update_frequency: 10
  publish_frequency: 10
  rolling_window: true
  width: 4.0
  height: 4.0
  resolution: 0.015
  transform_tolerance: 0.5

  plugins:
    - { name: static_layer, type: "costmap_2d::StaticLayer" }
    - { name: inflation_layer, type: "costmap_2d::InflationLayer" }
    - { name: obstacle_layer, type: "costmap_2d::ObstacleLayer" }

  static_layer:
    enabled: true
    map_topic: "/map"

  inflation_layer:
    enabled: true
    cost_scaling_factor: 5
    inflation_radius: 0.2

  obstacle_layer:
    observation_sources: laser_scan_sensor ir_sensor
    footprint_clearing_enabled: true
    track_unknown_space: true
    laser_scan_sensor:
      {
        data_type: LaserScan,
        sensor_frame: base_link,
        clearing: true,
        marking: true,
        obstacle_range: 3.0,
        raytrace_range: 3.5,
        observation_persistence: 2,
        topic: scan,
      }
    ir_sensor:
      {
        data_type: PointCloud,
        sensor_frame: base_link,
        clearing: true,
        marking: true,
        obstacle_range: 2,
        raytrace_range: 2.5,
        topic: distance_sensors,
      }
